rails/app/controllers/api/v1/external_activities_controller.rb

```diff
@@ -25,11 +25,18 @@ def create
     end
 
     external_activity = ExternalActivity.create(
-      :name               => name,
-      :url                => url,
-      :publication_status => params[:publication_status] || "private",
-      :user               => user,
-      :append_auth_token  => params[:append_auth_token] || false
+      :name                   => name,
+      :url                    => url,
+      :publication_status     => params[:publication_status] || "private",
+      :user                   => user,
+      :append_auth_token      => params[:append_auth_token] || false,
+      :author_url             => params[:author_url],
+      :print_url              => params[:print_url],
+      :tool                   => ActivityRuntimeAPI.create_tool(params[:source_type] || "LARA"),
```

Comment:
I just noticed this. It would be better if the source_type wasn't used in this API because I think it makes the API more specific to LARA. If just a tool url could be passed instead that seems better. 

However I haven't finished looking at the LARA side of this yet.

Reply:
I finished looking at LARA and it seems like the source_type could be avoided. Instead a tool_id would be sent. And then the appropriate tool would be looked up based on this tool_id which is really a URL. I'm not sure if this is called a tool_id in the portal code or something else.

The only question then is what to do if the tool doesn't exist yet. I guess the best option is to make a new method like ` ActivityRuntimeAPI.create_tool` which looks for the tool by url and then creates it if it doesn't exist. And has some hardcoded handling of the activity-player.concord.org url. Maybe we can find a better way to handle this in the future.

Reply:
@scytacki Tools have a `source_type` (described as "Used for the default reports."), a `name` (which seems to always match the source type, though it doesn't necessarily have to), and a `tool_id` (URL). 

If we don't send a `source_type` value, is it alright to leave that and the `name` blank for any new tool that gets created? It seems to work OK like that, but then in the portal settings form you end up with a drop-down menu for Tool that has an apparently blank option.

![image](https://user-images.githubusercontent.com/367459/100805165-30e4ac80-33fc-11eb-9740-2254b2577a6a.png)

Should a created tool be given an autogenerated name maybe?

Reply:
The automatic creation of the tool is only really useful during development or the initial setup of a new portal. 
My suggestion is that if the tool_id passed in is "https://activity-player.concord.org" and a tool with that id doesn't exist yet, then we create one with the name Activity Player and source type Activity Player.

I don't really remember all of the nuance of tools and sources.  
I'll spend some time documenting this. 
I see in your screenshot you have several tools for the activity player. Do you know why you have them? Is it to make testing something possible?

But I think you could try going ahead with the suggestion above.

Reply:
The multiple AP listings are an artifact of testing out the automatic creation of Tools. After one instance of automatically creating a new tool, I changed the tool's name, tool_id, and source_type values so I could create the same tool again. I forget exactly why now, but I wanted to keep the old one in case the new one didn't get created correctly. I also manually created the "Local Activity Player" tool. 

![image](https://user-images.githubusercontent.com/367459/101090022-045b9c80-3584-11eb-864c-9737c08bd08d.png)